### PR TITLE
sharing gl context between AWTGLCanvases for linux

### PR DESCRIPTION
**Pull request by hageldave**
This already existed for windows, this enables the `GLData.shareContext` attribute to be respected in `PlatformLinuxGLCanvas` as well.
refs #16 

I'm not quite sure if this is the right way to do it, but I tested it and it worked.

### CODE REVIEW

File: src/org/lwjgl/opengl/awt/PlatformLinuxGLCanvas.java [hunk 1]
```diff
@@ -48,7 +50,13 @@ private long create(int depth, GLData attribs, GLData effective) throws AWTExcep
 			// No framebuffer configurations supported!
 			throw new AWTException("No supported framebuffer configurations found");
 		}
-		long context = glXCreateNewContext(display, fbConfigs.get(0), GLX_RGBA_TYPE, NULL, true);
+		
+		long context;
+		if(attribs.shareContext != null){
+	    	context = glXCreateContext(display, glXGetVisualFromFBConfig(display, fbConfigs.get(0)), attribs.shareContext.context, true);
```
Comment (httpdigest) [inline src/org/lwjgl/opengl/awt/PlatformLinuxGLCanvas.java]: Why did you switch the whole function being used? Have you had a look at the actual parameters of [glXCreateNewContext](https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glXCreateNewContext.xml)? The fourth parameter is just the shared `GLXContext`.

Comment (hageldave) [reply src/org/lwjgl/opengl/awt/PlatformLinuxGLCanvas.java]: Oh i did not look into that, I was guessing from the name of the function that it was not capable of doing what i wanted (and the fact that it was not implemented yet even though being so trivial).

I'll fix that.

File: src/org/lwjgl/opengl/awt/PlatformLinuxGLCanvas.java [hunk 2]
```diff
@@ -48,7 +48,9 @@ private long create(int depth, GLData attribs, GLData effective) throws AWTExcep
 			// No framebuffer configurations supported!
 			throw new AWTException("No supported framebuffer configurations found");
 		}
-		long context = glXCreateNewContext(display, fbConfigs.get(0), GLX_RGBA_TYPE, NULL, true);
+		
+		long share_list = attribs.shareContext == null ? NULL:attribs.shareContext.context;
+		long context = glXCreateNewContext(display, fbConfigs.get(0), GLX_RGBA_TYPE, share_list, true);
```
Comment (hageldave) [inline src/org/lwjgl/opengl/awt/PlatformLinuxGLCanvas.java]: better this way?

**Comment by hageldave** [comment]
Hey there, I see there was another pull request that was conflicting with mine and that has recently been accepted. I just updated this PR to be compatible with the latest changes. I'm still interested in this feature for linux :)

**Comment by httpdigest** [comment]
Thank you for your work!

**Pull request merged**